### PR TITLE
Convert slash command data into a job factory instead of just a dictionary

### PR DESCRIPTION
**Pull request by sBuiltaJr**
This PR converts all slash commands using the job queue into a factory.  The factory removes special handling for certain kinds of jobs and will enable easier command expansion/integration in the future.

See the changelog for a more detailed description.

### CODE REVIEW

File: imageGenSD.py [hunk 1]
```diff
@@ -212,7 +213,37 @@ async def generate(interaction: dis.Interaction,
...
+        job = jf.JobFactory.GetJob(type=jf.JobTypeEnum.GENERATE,
+                                   ctx=interaction,
+                                   options=opts)
+        disLog.debug(f"Posting GENERATE job {job} to the queue.")
+        result = job_queue.Add(metadata=metadata,
+                               request=job)
+
+        await interaction.response.send_message(f'{result}', ephemeral=True, delete_after=9.0)
+        
+        """job = { 'metadata' : {
```
Comment (Mizuko) [inline imageGenSD.py:246]: Dead code in comment to be removed

File: src/managers/QueueMgr.py [hunk 2]
```diff
@@ -135,53 +119,55 @@ def Flush(self):
...
            Input: self - Pointer to the current object instance.
+                  metadata - Unpicklable data needed to post a result to Discord.
                   request - Sanitized data to potentially add to the queue.
 
            Output: str - Result of the job scheduling attempt.
         """
         global jobs
 
-        if request['data']['guild'] not in jobs:
+        if request.GetGuild() not in jobs:#request['data']['guild'] not in jobs:
```
Comment (Mizuko) [inline src/managers/QueueMgr.py:137]: Dead code in comment to be removed

File: src/utilities/TagRandomizer.py [hunk 3]
```diff
@@ -6,114 +6,128 @@
...
-        
+
         #Users trying to be cute will get the default result.
         if exact > 0:
             tag_count = exact
         else:
             tag_count = rand.randint(self.min_tags, self.max_tags)
-            
+
         for word in range(0, tag_count):
```
Comment (Mizuko) [inline src/utilities/TagRandomizer.py:91]: Fix indentation on `for` block from 13 spaces to 12 spaces

File: src/utilities/JobFactory.py [hunk 4]
```diff
@@ -0,0 +1,597 @@
+#encapsualtes different job types and post functions based on the specific
```
Comment (Mizuko) [inline src/utilities/JobFactory.py:1]: "Encapsulates"

File: src/utilities/JobFactory.py [hunk 5]
```diff
@@ -0,0 +1,597 @@
...
+import pickle as pic
+from . import ProfileGenerator as pg
+import requests as req
+from . import TagRandomizer as tr
+from typing import Optional
+from urllib.parse import urljoin
+
+#####  Package Variables  #####
+
+#####  Abstract Casses  #####
```
Comment (Mizuko) [inline src/utilities/JobFactory.py:24]: "Classes"

File: src/utilities/JobFactory.py [hunk 6]
```diff
@@ -0,0 +1,597 @@
...
+        """
+
+        return self.guild
+
+    def GetRandomize(self) -> bool:
+        """Returns whether this request should have randomized tags added.
+
+           Input: self - Pointer to the current object instance.
+
+           Output: bool - Whether or not the job needs ranomized tags added.
```
Comment (Mizuko) [inline src/utilities/JobFactory.py:58]: "randomized"

File: src/utilities/JobFactory.py [hunk 7]
```diff
@@ -0,0 +1,597 @@
...
+           For example, adds randomized tags to a request if a 'randomized'
+           flag is set true.
+
+           Input: self - Pointer to the current object instance.
+
+           Output: int - N/A.
+        """
+        pass
+
+#####  Enum Casses  #####
```
Comment (Mizuko) [inline src/utilities/JobFactory.py:119]: "Classes"

File: src/utilities/JobFactory.py [hunk 8]
```diff
@@ -0,0 +1,597 @@
...
+
+    GENERATE     =    0
+    ROLL         =    1
+    SHOWPROFILE  =    2
+    TESTPOST     =    3
+    TESTGET      =    4
+    TESTROLL     =    5
+    TESTSHOW     =    6
+
+#####  Job Casses  #####
```
Comment (Mizuko) [inline src/utilities/JobFactory.py:132]: "Classes"

**Review by Mizuko — CHANGES_REQUESTED** [review]
Consider a unified way to generate post/embed formatting, so Roll and ShowProfile don't have redundant code.

File: src/managers/QueueMgr.py (hunk 2 above)
Comment (sBuiltaJr) [reply src/managers/QueueMgr.py]: Corrected.

**Review by Mizuko — APPROVED** [review]

**Pull request merged**